src/core/createEvent.js

```diff
@@ -53,39 +117,7 @@ export default () => {
       return userXdm.web.webPageDetails.viewName;
     },
     toJSON() {
-      if (userXdm) {
-        event.mergeXdm(userXdm);
-      }
-      if (userData) {
-        content.data = userData;
-      }
-
-      if (lastChanceCallback) {
-        // We clone these because if lastChanceCallback throws an error, we don't
-        // want any modifications lastChanceCallback made to actually be applied.
-        const args = {
-          xdm: clone(content.xdm || {}),
-          data: clone(content.data || {})
-        };
-        try {
-          lastChanceCallback(args);
-          // If onBeforeEventSend throws an exception,
-          // we don't want to apply the changes it made
-          // so setting content.xdm and content.data is inside this try
-
-          // We only set content.xdm if content.xdm was already set or
-          // if content.xdm was empty and the lastChanceCallback added items to it.
-          if (content.xdm || !isEmptyObject(args.xdm)) {
-            content.xdm = args.xdm;
-          }
-          if (content.data || !isEmptyObject(args.data)) {
-            content.data = args.data;
-          }
-        } catch (e) {
-          // the callback should have already logged the exception
-        }
-      }
-
+      if (!isFinalized) this.finalize();
```

Comment:
If you want to guard against the event being used improperly, you could throw an error here instead. Silently running finalize without the onBeforeEventSend callback could be bad.

if (!isFinalized) {
  throw new Error("Attempt to call toJSON when the event has not yet been finalized.");
}